weave/trace_server/clickhouse_trace_server_batched.py

```diff
@@ -4573,6 +4573,18 @@ def _ch_call_dict_to_call_schema_dict(ch_call_dict: dict) -> dict:
     started_at = _ensure_datetimes_have_tz(ch_call_dict.get("started_at"))
     ended_at = _ensure_datetimes_have_tz(ch_call_dict.get("ended_at"))
     display_name = empty_str_to_none(ch_call_dict.get("display_name"))
+
+    # Load attributes from attributes_dump
+    attributes = _dict_dump_to_dict(ch_call_dict.get("attributes_dump", "{}"))
+
+    # For backwards compatibility: inject otel_dump into attributes if present
+    # Legacy trace servers stored all otel info in attributes, clients expect it
+    otel_dump = ch_call_dict.get("otel_dump")
+    if otel_dump:
+        otel_data = _dict_dump_to_dict(otel_dump)
+        if otel_data:
+            attributes["otel_span"] = otel_data
+
```

Comment:
returns a dict or raises, don't need to check if otel_data exists I don't think.
```
if (otel_dump := ch_call_dict.get("otel_dump")):
    attributes["otel_span"] = _dict_dump_to_dict(otel_dump)
```